demo/midgard-sdk/src/hub-oracle.ts

```diff
@@ -43,16 +48,87 @@ export const HubOracleDatumSchema = Data.Object({
 export type HubOracleDatum = Data.Static<typeof HubOracleDatumSchema>;
 export const HubOracleDatum = HubOracleDatumSchema as unknown as HubOracleDatum;
 
+export type HubOracleUTxO = {
+  utxo: UTxO;
+  datum: HubOracleDatum;
+  assetName: string;
+};
+
 /**
  * Parameters for the init transaction.
  */
-export type HubOracleInitParams = {};
+export type HubOracleInitParams = {
+  validator: AuthenticatedValidator;
+  datum: HubOracleDatum;
+};
 
 /**
  * Parameters for the burn transaction.
  */
 export type HubOracleBurnParams = {};
 
+export const getHubOracleDatumFromUTxO = (
+  nodeUTxO: UTxO,
+): Effect.Effect<HubOracleDatum, DataCoercionError> => {
+  const datumCBOR = nodeUTxO.datum;
+  if (datumCBOR) {
+    try {
+      const hubOracleDatum = Data.from(datumCBOR, HubOracleDatum);
+      return Effect.succeed(hubOracleDatum);
+    } catch (e) {
+      return Effect.fail(
+        new DataCoercionError({
+          message: `Could not coerce UTxO's datum to a hub oracle datum`,
+          cause: e,
+        }),
+      );
+    }
```

Comment:
It'd be desirable to use `Effect.try` instead of `try-catch` blocks for consistency.